### ISSUE TITLE
Rebuild for Apple Silicon

### ISSUE THREAD

**Issue by nickineering** [mosen/mysides#33]
It seems the version of MySides available on Homebrew does not support Apple Silicon. I have two Macs. It works fine on the Intel Mac, but on the Apple Silicon Mac I get the following error when trying to run `add`:

```
zsh: bad CPU type in executable: mysides
```

This is an amazingly helpful project, but seeing the way Apple is moving it would be really nice if it were republished supporting the latest processors.

Thank you!

**Comment by acsr** [comment]
I cannot confirm this issue on Ventura M1 Pro Max. I use version 1.0.1 from 26. Sept 2016 (49356 bytes)
see: https://github.com/mosen/mysides/issues/32  for (coarse) checking installed version.

In general it work except for segmentation fault errors on certain items. 
see: https://github.com/mosen/mysides/issues/29 on this and how to kind of workaround.

**Comment by neoighodaro** [comment]
> I cannot confirm this issue on Ventura M1 Pro Max. I use version 1.0.1 from 26. Sept 2016 (49356 bytes) see: #32 for (coarse) checking installed version.
> 
> In general it work except for segmentation fault errors on certain items. see: #29 on this and how to kind of workaround.

It'll work if you have Rosetta but new macs dont have this by default so would be great to get a native build

**Comment by mosen** [comment]
I do apologise I have not looked at the project in years but I can only promise to give it a shot at least. Got a new silicon mac from work so it's theoretically possible for me to test now

**Comment by neoighodaro** [comment]
That would be amazing! Really appreciate your work on it! I considered building myself but wasn’t sure if it’d be possible especially since I don’t know much about the project myself

**Comment by hle-pc** [comment]
Yay. Once this is updated, it could be re-added to Homebrew. It was recently deprecated from there due to lack of maintenance.

**Comment by dantveita** [comment]
Any update on this?

**Comment by huyz** [comment]
To be honest, running mysides is such a very infrequent operation (backup and restore) that relying on Rosetta works just fine. And I have to restore more frequently than you'd think because of macOS bugs!

**Comment by acsr** [comment]
If you want to build this yourself for Apple Silicon, take a look into the forks. Its is on my worklist, but very far from high priority. 1.01 is still working for me. The only thing that breaks listing are aliases / links pointing to servers not mounted during listing.

So I wrote a script to automate the export to have dumps handy if something breaks.

```` shell
#!/bin/bash
############################
SCRIPTNAME="DumpMySidebar.sh"
############################

VERSION="V1.0.0 20250318_132059-acsr"

# Installed at /Users/Shared/GlobalDefinesZentral from   ## (internal comment)
# Needs proper permissions: 'chmod 754 DumpMySidebar.sh``

DATETIMESTRING=`date +%Y%m%d_%H%M%S`
MACHINE="MyMachinename"  # for creating filename
DESTINATIONFOLDER="/Users/[MyUser]/Documents/"
EXPORTFILENAME=$DATETIMESTRING-$MACHINE-mysides-list.txt
EXPORTFILEPATH=$DESTINATIONFOLDER/$EXPORTFILENAME
EXPORTCOMMAND="mysides list >> '"$EXPORTFILEPATH"'"

echo
echo ">> "$SCRIPTNAME", Version: "$VERSION

#write into EXPORTFILE HEADER
echo >> "$EXPORTFILEPATH" "### "$DATETIMESTRING" Exporting "$MACHINE" Sidebar items to file using mysides command"
echo >> "$EXPORTFILEPATH" "### requires manual install of mysides command v1.0.1(https://github.com/mosen/mysides/releases/tag/v1.0.1)"
echo >> "$EXPORTFILEPATH" "### DumpMySidebar.sh V1.0.0 20250318_132059-acsr\n"
echo >> "$EXPORTFILEPATH" "# To create script for restore use mysides syntax:"
echo >> "$EXPORTFILEPATH" "# Usage: mysides list|add <name> <uri>|remove <name>\n"
echo >> "$EXPORTFILEPATH" "# 	 list - list sidebar items"
echo >> "$EXPORTFILEPATH" "# 	 add - append a sidebar item to the end of the list"
echo >> "$EXPORTFILEPATH" "# 	 remove - remove a sidebar item\n#"
echo >> "$EXPORTFILEPATH" "# Output\n"

# Run Export Command
echo ">> command: "$EXPORTCOMMAND
eval $EXPORTCOMMAND

# open Exportfile
echo ">> opening file"
open "$EXPORTFILEPATH"

# open Export Destination Folder
echo ">> show destination folder in finder"
open "$DESTINATIONFOLDER"

````

My Todos:

- [x] Create local forks and archives of this repo and all forks (I was hit by code, removed from github hard in the past)
- [ ] Create remove script code from listing
- [ ] Create a recreate script code from listing
- [ ] Create a diff report 
    - [ ] (Hard dumb way: first list, then remove latest dump, then list changes, remove all, recreate)
    - [ ] Elegant: compare last dump with current dump for diff report
- [ ] Recompile for Apple Silicon

**Comment by acsr** [comment]
Addendum I use my text editor to reformat the output of the dump script to a remove/create script using multiline selection/edit

**Comment by mosen** [comment]
Yeah sorry still intending to do it but main job not apple related. Will try to make some time in next couple weeks.
